### ISSUE TITLE
Swagger Validation for File Downloads

### ISSUE THREAD

**Issue by Srithar-ccs** [apigee-127/swagger-tools#272]
My Code is like this:
```
			var archive = archiver('zip');			
			var totalAcitemListArray = totalAcitemList.toArray();			
			for (var index = 0; index < totalAcitemListArray.length; index++) {
				archive.append(fs.createReadStream(path.join(root_directory, totalAcitemListArray[index])), { name:totalAcitemListArray[index] });
			}			
			archive.finalize();
			archive.on('error', function(err) {
				response.status(500).send({error: err.message});
			});
			response.send('acitemsPackage.zip');
```
and in the YAML file i have also declared like this:

```
produces:
  - application/zip

/library/getacitemspackage:
  # This is the controller for getting the acitem and all its internal references as a deep package for the seamless delivery
    x-swagger-router-controller: library_get_acitems_package
    # this is a post method verb of the api service call
    post:
      summary: The API does the zip of all the given acitems and brings all the deep copy references packages from the cloud library
      description: The API does the zip of all the given acitems and brings all the deep copy references packages from the cloud library
      operationId: libraryGetAcitemsPackage
      parameters:
        - name: acitemsList
          in: body
          description: The list of all the acitems for which the package needs to be delivered as a zip
          schema:
            type: object
      responses:
        "200":
          description: Success
          schema:            
            type: object
        default:
          description: Error
          schema:
            $ref: "#/definitions/ErrorResponse"
```

```
SyntaxError: Response validation failed: value expected to be an array/object but is not
    at Object.parse (native)
````
I tried with diff schema types but the response is still not obtained.
Also the Content Type is an attachment

**Comment by whitlockjc** [comment]
`type` `object` is only for JSON objects and I'm sure your ZIP file is not a JavaScript object.  Based on your usage of `response.send`, it's actually a `string` with the content of `'acitemsPackage.zip'`.  So the error is as expected.  Also, if you plan on sending a ZIP file, I don't think `object` is the proper `type`.  It will likely be `string` or something like that.  I'll keep this open to give you a chance to respond.

**Comment by Srithar-ccs** [comment]
oh sorry @whitlockjc , since i was trying out different things i had to try with type as object. The issue goes like this, i followed the exact steps in the swagger-node page https://github.com/apigee-127/swagger-node, after that i changed the response schema type to file, with the response.attachment 
```
var archive = archiver('zip');
			var totalAcitemListArray = totalAcitemList.toArray();			
			for (var index = 0; index < totalAcitemListArray.length; index++) {
				archive.append(fs.createReadStream(path.join(root_directory, totalAcitemListArray[index])), { name:totalAcitemListArray[index] });
				
			}			
			archive.finalize();
			archive.on('error', function(err) {
				response.status(500).send({error: err.message});
			});
			response.attachment('acitemsPackage.zip');			
			archive.pipe(response);
```

YAML file:
```
responses:
        "200":
          description: Success
          schema:            
            type: file
```
It shows as the type not available in anyof ......

For more details:
https://github.com/swagger-api/swagger-node/issues/283

Basically the swagger-node needs a update for the swagger-tools

**Comment by Srithar-ccs** [comment]
https://github.com/swagger-api/swagger-node/issues/283

This issue has been resolved by @webron and @theganyo. Closing this issue. Thanks a lot for your help.
